rust/agama-cli/src/config.rs

```diff
@@ -103,6 +216,141 @@ pub async fn run(
     }
 }
 
+/// Validate a JSON profile, by doing a HTTP client request.
+async fn validate_client(
+    client: &BaseHTTPClient,
+    url_or_path: CliInput,
+) -> anyhow::Result<ValidationOutcome> {
+    let mut url = client.base_url.join("profile/validate").unwrap();
+    url_or_path.add_query(&mut url)?;
+
+    let body = url_or_path.body_for_web()?;
+    // we use plain text .body instead of .json
+    let response: Result<reqwest::Response, agama_lib::error::ServiceError> = client
+        .client
+        .request(reqwest::Method::POST, url)
+        .body(body)
+        .send()
+        .await
+        .map_err(|e| e.into());
+
+    let result = client.deserialize_or_error(response?).await;
+    result.map_err(|e| e.into())
+}
+
+async fn validate(client: &BaseHTTPClient, url_or_path: CliInput) -> anyhow::Result<()> {
+    let validity = validate_client(client, url_or_path).await?;
+    match validity {
+        ValidationOutcome::Valid => {
+            eprintln!("{} {}", style("\u{2713}").bold().green(), validity);
+        }
+        ValidationOutcome::NotValid(_) => {
+            eprintln!("{} {}", style("\u{2717}").bold().red(), validity);
+        }
+    }
+    Ok(())
+}
+
+fn is_autoyast(url_or_path: &CliInput) -> bool {
+    let path = match url_or_path {
+        CliInput::Path(pathbuf) => pathbuf.as_os_str().to_str().unwrap_or_default().to_string(),
+        CliInput::Url(url_string) => {
+            let url = Uri::parse(url_string.as_str()).unwrap_or_default();
+            let path = url.path().to_string();
+            path
+        }
+        _ => {
+            return false;
+        }
+    };
+
+    path.ends_with(".xml") || path.ends_with(".erb") || path.ends_with('/')
+}
+
+async fn generate(client: &BaseHTTPClient, url_or_path: CliInput) -> anyhow::Result<()> {
```

Comment:
The `generate` command needs to resolve the relative URLs. I guess you should parse the JSON into an `InstallSettings` struct and call the `resolve_url` function to get the job done.